### PR TITLE
District page

### PR DESCRIPTION
**Pull request by andrewjtimmons**
what?
return stats on number users, voters and letters for a district if the param `get_adoption_details: true` is set for lookup-district

sample response
```
[{"num_users_using_district":"1","voters_available":"10","voters_adopted":"0","letters_prepped":"0","letters_sent":"0","id":1,"district_id":"AZ02","state":"Arizona","state_abbr":"AZ","district_num":"02","description":"WORDSWORDSWORDS","lat":32.61,"long":-110.48,"coordinates":{"x":32.61,"y":-110.48},"return_address":"1517 N Wilmot Rd #121","ra_city":"Tucson","ra_state":"AZ","ra_zip":"85712"}]
```

testing:
tested in postgres and on dev.  Changed the api call for looking up a district in dashboard.js to be 
```
       url: `${process.env.REACT_APP_API_URL}/lookup-district`,
        params: {
          district_id: districtId,
          get_adoption_details: true
          }
```
and looked at the response in the browser

Also checked that knex escapes these parameters https://knexjs.org/#Raw and it does

Also tested rate limits and they work.

risks:
low risk
1.  this makes the district info api not require auth.  it is now rate limited to 10 calls per minute
2.  this makes us run a multi table join.  if perf gets bad it could be bad. probably fine.

future tech debt thoughts
this might be better as separate functions, but callback and promise changing heck made us go this way.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -369,17 +369,76 @@ router.route('/get-districts')
...
+                    else 0
+                end as sent
+                from voters
+                where district_id = ?
+            ) voters_case
+            group by district_id
+        ) voters_agg
+        on districts.district_id = voters_agg.district_id
+        where districts.district_id = ?
+        group by 2,3,4,5,6;`,
```
[the remark this answers is not in the document]
Comment (andrewjtimmons) [reply server/server.js]: fixed https://github.com/sjforman/votefwd/pull/138/commits/703b3497122bf5aa277625c8249e56b6e1d21d48

File: server/server.js [hunk 2]
```diff
@@ -369,17 +369,76 @@ router.route('/get-districts')
...
-      .where({ district_id: req.query.district_id })
+  .get(rateLimits.lookupDistrictRateLimit, function(req, res) {
+    if (JSON.parse(req.query.get_adoption_details) == true){
+      db.raw(`select
+        count(users.id) as num_users_using_district,
+        voters_agg.voters_available,
+        voters_agg.voters_adopted,
+        voters_agg.letters_prepped,
+        voters_agg.letters_sent,
+        districts.*
```
[the remark this answers is not in the document]
Comment (andrewjtimmons) [reply server/server.js]: fixed https://github.com/sjforman/votefwd/pull/138/commits/703b3497122bf5aa277625c8249e56b6e1d21d48

**Comment by andrewjtimmons** [comment]
TODO before merge
add indexes on DBs

**Comment by andrewjtimmons** [comment]
RE testing there is not an existing test.  Server tests are currently `xit`ed.  I believe this was because we did not have mock creds.  I chatted with Scott about moving this into a function and fixtures and he said lets skip for now given the time crunch.

**Comment by andrewjtimmons** [comment]
Actually this api is unauthed so we could add a test.  However axios sends them in a key called query and requests sends them in a key called param.  I am going to make a github issue to make tests for this for myself to do later but want to merge this to unblock garrett

**Comment by andrewjtimmons** [comment]
Added indexes to the DBs.  Voters got one for `district_id` which is the where clause of the query.  Users got one for `current_district` which is the join key for the query.

**Comment by andrewjtimmons** [comment]
Made todo for test https://github.com/sjforman/votefwd/projects/1#card-13036055

**Comment by yoni** [comment]
LGTM

**Pull request merged**